protob/messages.proto

```diff
@@ -912,6 +922,102 @@ message CosiSignature {
 	optional bytes signature = 1;		// Signature
 }
 
+///////////////////
+// Iota messages //
+///////////////////
+
+/**
+ * Request: Show seed on the device
+ * @next Success
+ * @next Failure
+ */
+message IotaShowSeed {
+}
+
+/**
+ * Request: Get address counter
+ * @next IotaGetAddressCounter
+ */
+message IotaGetAddressCounter {
+}
+
+/**
+ * Returns the address counter as currently set in storage
+ * @prev IotaGetAddressCounter
+ */
+message IotaAddressCounter {
+	required uint32 address_counter = 1;
+}
+
+/**
+ * Request: Set address counter to storage
+ * @next Success
+ * @next Failure
+ */
+message IotaSetAddressCounter {
+	required uint32 address_counter = 1;
+}
+
+/**
+ * Request: Ask device for a public address
+ * @next IotaAddress
+ */
+message IotaGetAddress {
+	optional uint32 address_index = 1;
+}
+
+/**
+ * Response: Contains public address
+ * @prev IotaGetAddress
+ */
+message IotaAddress {
+	required uint32 address_index = 1;
+	required string address = 2;
+}
+
+/**
+ * Request: Approve a value transfer.
+ * @next IotaTxApproved
+ * @next Failure
+ */
+message IotaTxRequest {
+	required string receiving_address = 1;			// Address to which the amount is send
+	required uint64 transfer_amount = 2;			// Amount of iota to be transferred
+	required uint64 balance = 3;					// Balance on the current address (to calculate remainder)
+    required uint64 request_timestamp = 4;			// Final transaction must be within certain time from here
+	optional uint32 input_address_index = 5;		// Seed index from where to sent
+	optional uint32 remainder_address_index = 6;	// Seed index for the remaining balance
+	optional string tag = 7;						// Transaction tag, defaults to 'TREZOR'
+}
+
+/**
+ * Response: Transaction is allowed.
+ * This triggers the computer to send the current timestamp
+ * @next IotaTxDetails
+ * @next Failure
+ */
+message IotaTxApproved {
+}
+
+/**
+ * Request: Send timestamp to initiate transaction signing.
+ * @next IotSignedTx
+ * @next Failure
+ */
+message IotaTxDetails {
```

Comment:
Why does this need to be separate from `IotaTxRequest`?

Reply:
Tried to explain it in https://github.com/trezor/trezor-mcu/pull/255